pkg/virt-controller/watch/vm.go

```diff
@@ -2242,8 +2242,12 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 	var syncErr syncError
 	var err error
 
+	if !c.needsSync(key) {
```

Comment:
Was there a need to move this?

Reply:
Not exactly, I just thought that an early exit should have gone earlier.